subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,303 @@
+// tag::maven-ivy-repository-no-auth[]
+repositories {
+    maven {
+        url = uri("http://repo.mycompany.com/maven2")
+    }
+
+    ivy {
+        url = uri("http://repo.mycompany.com/repo")
+    }
+}
+// end::maven-ivy-repository-no-auth[]
+
+// tag::maven-ivy-repository-auth[]
+repositories {
+    maven {
+        url = uri("sftp://repo.mycompany.com:22/maven2")
+        credentials {
+            username = "user"
+            password = "password"
+        }
+    }
+
+    ivy {
+        url = uri("sftp://repo.mycompany.com:22/repo")
+        credentials {
+            username = "user"
+            password = "password"
+        }
+    }
+}
+// end::maven-ivy-repository-auth[]
+
+// tag::maven-ivy-s3-repository[]
+repositories {
+    maven {
+        url = uri("s3://myCompanyBucket/maven2")
+        credentials(AwsCredentials::class.java) {
```

Comment:
:x: please use `credentials(AwsCredentials::class) {` instead, and in other places in this PR

IntelliJ still wrongly highlights this as an error but it's already fixed in Kotlin 1.2.70 development builds, see https://github.com/gradle/kotlin-dsl/issues/1064